### PR TITLE
release: update minor release issue template

### PR DESCRIPTION
**Pull request by unknwon**
Various formatting, links, and wording updates to make intention clear.

## Test plan

Only changes to release issue template

---

Derived from my [retro notes for the 3.40 release](https://docs.google.com/document/d/1VHBqsWocffw9W8NRGhw4lLzyaoqgc1fiNdmTMMaVLxM/edit#).

### CODE REVIEW

File: content/departments/engineering/dev/process/releases/release_issue_template.md [hunk 1]
```diff
@@ -108,8 +110,12 @@ On the day of the release, confirm there are no more release-blocking issues (as
 
 ### Post-release
 
-- [ ] Notify the next [release captain](./index.md#release-captain) that they are on duty for the next release.
```
Comment (unknwon) [inline content/departments/engineering/dev/process/releases/release_issue_template.md:111]: This link is broken, but using an absolute link failed on lint, i'd rather just remove it.

File: content/departments/engineering/dev/process/releases/release_issue_template.md [hunk 2]
```diff
@@ -94,12 +96,12 @@ On the day of the release, confirm there are no more release-blocking issues (as
...
 - [ ] From the [release batch change](https://k8s.sgdev.org/organizations/sourcegraph/batch-changes), merge the release-publishing PRs created previously.
+  - For [sourcegraph](https://github.com/sourcegraph/sourcegraph)
+    - [ ] Cherry pick the release-publishing PR from `sourcegraph/sourcegraph@main` into the release branch.
   - For [deploy-sourcegraph](https://github.com/sourcegraph/deploy-sourcegraph)
-    - [ ] Ensure the [release](https://github.com/sourcegraph/deploy-sourcegraph/releases) has the correct tag
-- [ ] For [deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker)
-  - [ ] Ensure the [release](https://github.com/sourcegraph/deploy-sourcegraph-docker/releases) has the correct tag
-- For [sourcegraph](https://github.com/sourcegraph/sourcegraph), also:
-  - [ ] Cherry pick the release-publishing PR from `sourcegraph/sourcegraph@main` into the release branch.
+    - [ ] Ensure the [release tag `v$MAJOR.$MINOR.$PATCH`](https://github.com/sourcegraph/deploy-sourcegraph/releases) has been created
```
Comment (caugustus-sourcegraph) [inline content/departments/engineering/dev/process/releases/release_issue_template.md]:
```suggestion
    - [ ] Ensure the [release tag `v$MAJOR.$MINOR.$PATCH`](https://github.com/sourcegraph/deploy-sourcegraph/tags) has been created
```

File: content/departments/engineering/dev/process/releases/release_issue_template.md [hunk 3]
```diff
@@ -94,12 +96,12 @@ On the day of the release, confirm there are no more release-blocking issues (as
...
+  - For [deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker)
+    - [ ] Ensure the [release tag `v$MAJOR.$MINOR.$PATCH`](https://github.com/sourcegraph/deploy-sourcegraph-docker/releases) has been created
```
Comment (caugustus-sourcegraph) [inline content/departments/engineering/dev/process/releases/release_issue_template.md]:
```suggestion
    - [ ] Ensure the [release tag `v$MAJOR.$MINOR.$PATCH`](https://github.com/sourcegraph/deploy-sourcegraph-docker/tags) has been created
```

**Review by caugustus-sourcegraph — APPROVED** [review]

**Review by kevinwojo — APPROVED** [review]
+1 to Crystal's suggestions, otherwise 👍

**Pull request merged**